ProcessMaker/Repositories/CaseUtils.php

```diff
@@ -31,72 +31,105 @@ public static function getCaseNumberByKeywords(int $caseNumber)
 
     /**
      * Store requests.
      *
-     * @param ExecutionInstanceInterface $instance
      * @param Collection $requests
+     * @param array|null $requestData
+     *     An optional array of additional request data. Each element in the array should have the following structure:
+     *     [
+     *         'id' => int,                     // The unique identifier of the request
+     *         'name' => string,                // The name of the request
+     *         'parent_request_id' => int|null  // The ID of the parent request, or null if there is no parent
+     *     ]
      * @return Collection
      */
-    public static function storeRequests(ExecutionInstanceInterface $instance, Collection $requests)
+    public static function storeRequests(Collection $requests, ?array $requestData = []): Collection
     {
-        return $requests->push([
-            'id' => $instance->id,
-            'name' => $instance->name,
-            'parent_request_id' => $instance?->parentRequest?->id,
-        ])
-        ->unique('id')
-        ->values();
+        // check if the request data has the required keys
+        if (!empty($requestData) && array_key_exists('id', $requestData) && array_key_exists('name', $requestData)) {
+            $requests->push($requestData);
+        }
+
+        return $requests->unique('id')->values();
     }
 
     /**
      * Store request tokens.
      *
-     * @param int $tokenId
      * @param Collection $requestTokens
+     * @param int|null $tokenId
      * @return Collection
      */
-    public static function storeRequestTokens(int $tokenId, Collection $requestTokens)
+    public static function storeRequestTokens(Collection $requestTokens, ?int $tokenId = null): Collection
     {
-        return $requestTokens->push($tokenId)
-            ->unique()
-            ->values();
+        if (!is_null($tokenId)) {
+            $requestTokens->push($tokenId);
+        }
+
+        return $requestTokens->unique()->values();
     }
 
     /**
      * Store tasks.
      *
-     * @param TokenInterface $token
      * @param Collection $tasks
+     * @param array|null $taskData
+     *     An optional array of additional task data. Each element in the array should have the following structure:
+     *     [
+     *         'id' => int,         // The unique identifier of the task
+     *         'element_id' => int, // The unique identifier of the element
+     *         'name' => string,    // The name of the element
+     *         'process_id' => int  // The unique identifier of the process
+     *         'element_type' => string  // The type of the element
+     *     ]
+     * @return Collection
+     */
+    public static function storeTasks(Collection $tasks, ?array $taskData = []): Collection
+    {
+        if (!empty($taskData) && array_key_exists('id', $taskData) && array_key_exists('element_id', $taskData) && array_key_exists('name', $taskData) && array_key_exists('process_id', $taskData)) {
```

Comment:
review if is better add this after this if block to avoid unnecessary verifications
 
  if (in_array($taskData['element_type'], self::ALLOWED_ELEMENT_TYPES)) {

Reply:
It was fixed